### DISCUSSION TITLE
A basic RJSON file with minimal abstracted config

### DISCUSSION THREAD

**Discussion by Himujjal** [abenz1267/neoconf#3] [Ideas]
Continuing on our discussion in #1 . RJSON is just a file format. Its an alternative for YAML, JSON, TOML.

What the config file will do is take away the pain of writing lua files and provide project management settings. Like the language servers that are required to be installed, the tree-sitter plugins to be installed and other plugins too. A simple `neoconf -f config.rjson` will result in a generated configuration with all the hooks and crooks needed.

**Why will this be beneficial?**

Switching machines. run `curl <url_to_config>.rjson | neoconf -f` to set up everything from installing neovim distribution to generating lua configs.

**Comment by abenz1267** [comment]
Yes, this is the goal... But why not just use JSON? It’s a universally accepted format most developers know, we don’t need to introduce a dependency and the end user should touch that file anyways.

**Reply by Himujjal** [reply]
Well. JSON can be used. I agree. Scrap the RJSON idea

We can keep this discussion for the minimal config though
